kadi/commands/observations.py

```diff
@@ -0,0 +1,395 @@
+    with the same obsid, so this function always returns a list of observation
+    parameters even when ``obsid`` is specified. This most frequently occurs
+    after any unexpected stoppage of the observng loads (SCS-107) which
+    therefore cancels subsequent obsid commanding. In many cases you can just
+    use the first element.
+
+    Examples::
+
+        >>> from kadi.commands import get_observations
+        >>> obs = get_observations(obsid=8008)[0]
+        >>> obs
+        {'obsid': 8008,
+        'simpos': 92904,
+        'obs_stop': '2007:002:18:04:28.965',
+        'manvr_start': '2007:002:04:31:48.216',
+        'targ_att': (0.149614271, 0.490896707, 0.831470649, 0.21282047),
+        'npnt_enab': True,
+        'obs_start': '2007:002:04:46:58.056',
+        'prev_att': (0.319214732, 0.535685207, 0.766039803, 0.155969017),
+        'starcat_idx': 144398,
+        'source': 'DEC2506C'}
+
+        >>> obs_all = get_observations()  # All observations in commands archive
+
+        # Might be convenient to handle this as a Table
+        >>> from astropy.table import Table
+        >>> obs_all = Table(obs_all)
+
+    :param obsid: int, None
+        ObsID
+    :param start: CxoTime-like, None
+        Start time (default=beginning of commands)
+    :param stop: CxoTime-like, None
+        Stop time (default=end of commands)
+    :param scenario: str, None
+        Scenario
+    :param cmds: CommandTable, None
+        Use this command table instead of querying the archive.
+    :returns: list of dict
+        Observation parameters for matching observations.
+    """
+    from kadi.commands.commands_v2 import get_cmds
+    start = CxoTime('1999:001' if start is None else start)
+    stop = (CxoTime.now() + 1 * u.year) if stop is None else CxoTime(stop)
+
+    if cmds is None:
+        if scenario not in OBSERVATIONS:
+            cmds = get_cmds(scenario=scenario)
+            cmds_obs = cmds[cmds['tlmsid'] == 'OBS']
+            obsids = []
+            for cmd in cmds_obs:
+                if cmd['params'] is None:
+                    _obsid = cmd['obsid']
+                else:
+                    _obsid = cmd['params']['obsid']
+                obsids.append(_obsid)
+
+            cmds_obs['obsid'] = obsids
+            OBSERVATIONS[scenario] = cmds_obs
+        else:
+            cmds_obs = OBSERVATIONS[scenario]
+    else:
+        cmds_obs = cmds[cmds['tlmsid'] == 'OBS']
+
+    i0, i1 = cmds_obs.find_date([start.date, stop.date])
+    cmds_obs = cmds_obs[i0:i1]
+
+    if obsid is not None:
+        cmds_obs = cmds_obs[cmds_obs['obsid'] == obsid]
+        if len(cmds_obs) == 0:
+            raise ValueError(f'No matching observations for {obsid=}')
+
+    obss = [cmd['params'] for cmd in cmds_obs]
+    for obs, cmd_obs in zip(obss, cmds_obs):
+        obs['source'] = cmd_obs['source']
+
+    return obss
+
+
+def get_agasc_cone_fast(ra, dec, radius=1.5, date=None):
```

Comment:
Should this just move into `agasc`?

Reply:
It's getting late, so unless this is explicitly wrong or bad I'd like to leave it. Maybe when we come across a 2nd application that needs this function we can refactor into the core agasc.